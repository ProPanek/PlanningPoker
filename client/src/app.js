import React, { Component } from "react";
import {inject, observer, Provider} from "mobx-react";
import { withRouter } from "react-router-dom";
import { Grid } from "@material-ui/core";
import Header from "./components/static/header";
import Router from "./components/router/router";
import Notification from "./components/notification/notification"

class App extends Component {
  render() {
    return (
      <div className="App">
        <Notification />
        <Header />
          <Grid container>
           <Router />
          </Grid>
      </div>
    );
  }
}
export {App};
export default inject("store")(withRouter(observer(App)));

