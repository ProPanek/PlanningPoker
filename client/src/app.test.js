import React from "react";
import {App} from "./app";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

const noop = () => {}
it("renders without crashing", () => {
    const store = {
        fetchRooms: noop
    }
    const component = shallow(<App store={store} />)
    expect(component).toMatchSnapshot()
});
