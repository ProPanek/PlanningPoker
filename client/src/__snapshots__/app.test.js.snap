// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    store={
      Object {
        "fetchRooms": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <inject-Notification-with-store />,
        <inject-withRouter(Header)-with-store />,
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <WithStyles(Grid)
            container={true}
          >
            <inject-withRouter(Router)-with-store />
          </WithStyles(Grid)>
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Grid)
            container={true}
          >
            <inject-withRouter(Router)-with-store />
          </WithStyles(Grid)>,
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <inject-withRouter(Router)-with-store />,
            "container": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <inject-Notification-with-store />,
          <inject-withRouter(Header)-with-store />,
          <div
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <WithStyles(Grid)
              container={true}
            >
              <inject-withRouter(Router)-with-store />
            </WithStyles(Grid)>
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <WithStyles(Grid)
              container={true}
            >
              <inject-withRouter(Router)-with-store />
            </WithStyles(Grid)>,
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <inject-withRouter(Router)-with-store />,
              "container": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
